core/src/main/scala/org/http4s/util/package.scala

```diff
@@ -54,4 +54,23 @@ package object util {
     "Moved to org.http4s.execution.trampoline, is now merely a ExecutionContextExecutor.",
     "0.18.0-M2")
   val TrampolineExecutionContext: ExecutionContextExecutor = execution.trampoline
+
+  /* This is nearly identical to the hashCode of java.lang.String, but converting
+   * to lower case on the fly to avoid copying `value`'s character storage.
+   */
+  def hashLower(s: String): Int = {
```

Comment:
I hope this is still lightning fast since I moved it to a function.  I assume yes, but haven't measured.